.github/workflows/publish_website.yml

```diff
@@ -34,8 +34,17 @@ jobs:
         uses: actions/setup-python@7f80679172b057fc5e90d70d197929d454754a5a # v2
         with:
           python-version: 3.x
-      - name: Installing Material for MkDocs
-        run: pip install mkdocs-material
+          cache: 'pip'
+      - name: Installing dependencies
+        run: |
+          pip install \
+            mkdocs-material \
+            requests
+      - name: Generating release index
+        run: |
+          build_tools/scripts/generate_release_index.py \
+            --owner="${GITHUB_REPOSITORY}" \
+            --output=docs/website/overrides/pip-release-links.html
```

Comment:
> We can incorporate this into our docs pipeline and upload it to the
website.

I'd prefer to add a new subdomain to iree.dev (`perf.iree.dev`, `docs.iree.dev`, `iree.dev/releases/` or `releases.iree.dev`, etc.), but I guess we could add this HTML file to https://iree-org.github.io/iree/ as https://iree-org.github.io/iree/package-index/ ...

For that, we'd need to either

(A) (what you have here) fit the file(s) under [docs/website/](https://github.com/iree-org/iree/tree/main/docs/website) somehow, prior to https://github.com/iree-org/iree/blob/9a0058e32d92f549e72f7d4b8e0fb98275587c5c/.github/workflows/publish_website.yml#L43-L44 

(B) Follow that step with a git push to https://github.com/iree-org/iree/tree/gh-pages including the new file(s)

(C) Unpack the `mkdocs gh-deploy` command (https://www.mkdocs.org/user-guide/deploying-your-docs/#project-pages) to build, add the new file(s), then push

Adding a new subdomain and publishing files to it is apparently really tricky with how we manage domains though... so this is probably the best we can do easily :/

Reply:
We haven't even got `docs.iree.dev` working, so I don't really want to block on that... I don't think this needs to live where we put it now forever

Reply:
Agreed, though there is some work involved for downstream users (and us) whenever a URL like this changes.